rest/data/auth.go

```diff
@@ -0,0 +1,91 @@
+package data
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"github.com/evergreen-ci/cedar/util"
+	"github.com/evergreen-ci/gimlet"
+	"github.com/jpillora/backoff"
+	"github.com/pkg/errors"
+)
+
+/////////////////////////////
+// DBConnector Implementation
+/////////////////////////////
+
+// EvergreenProxyAuthLogRead sends a http request to Evergreen to check if the
+// given user is authorized to read the log belonging to the given project. If
+// the user is authorized, nil is returned, otherwise a gimlet.Responder with
+// the appropriate http error code is returned.
+func (dbc *DBConnector) EvergreenProxyAuthLogRead(ctx context.Context, userToken, baseURL, resourceId string) gimlet.Responder {
+	urlString := fmt.Sprintf("%s?resource=%s&resource_type=project&permission=project_logs&required_level=10", baseURL, resourceId)
+	req, err := http.NewRequest(http.MethodGet, urlString, nil)
+	if err != nil {
+		return gimlet.MakeJSONInternalErrorResponder(errors.Wrap(err, "error creating http request"))
+	}
+	req = req.WithContext(ctx)
+	req.Header = map[string][]string{"mci-token": {userToken}}
```

Comment:
I think there's a method to add a key to the header that's better to use than the map itself. 

also put "mci-token" in a constant somewwhere.

Reply:
should it be a global constant?